network/discovery.go

```diff
@@ -246,22 +264,103 @@ func (d *discovery) findPeersCall(peerID peer.ID) ([]*peer.AddrInfo, error) {
 }
 
 func (d *discovery) run() {
+	ticker1 := time.NewTicker(peerDiscoveryInterval)
+	ticker2 := time.NewTicker(bootnodeDiscoveryInterval)
+
+	defer func() {
+		ticker1.Stop()
+		ticker2.Stop()
+	}()
+
 	for {
 		select {
-		case <-time.After(5 * time.Second):
-		case <-d.notifyCh:
 		case <-d.closeCh:
 			return
+		case <-ticker1.C:
+			go d.handleDiscovery()
+		case <-ticker2.C:
+			go d.bootnodeDiscovery()
 		}
-		d.handleDiscovery()
 	}
 }
 
 func (d *discovery) handleDiscovery() {
 	// take a random peer and find peers
-	if target := d.peers.getRandomPeer(); target != nil {
-		if err := d.attemptToFindPeers(target.id); err != nil {
-			d.srv.logger.Error("failed to dial peer", "peer", target.id, "err", err)
+	if d.srv.numOpenSlots() > 0 {
+		if target := d.peers.getRandomPeer(); target != nil {
+			if err := d.attemptToFindPeers(target.id); err != nil {
+				d.srv.logger.Error("failed to find new peers", "peer", target.id, "err", err)
+			}
+		}
+	}
+}
+
+// bootnodeDiscovery queries a random bootnode for new peers and adds them to the routing table
+func (d *discovery) bootnodeDiscovery() {
+	if d.srv.numOpenSlots() <= 0 {
+		return
+	}
+
+	// get a random bootnode which is not connected
+	bootNode := d.srv.getBootNode()
+	if bootNode == nil {
+		return
+	}
+
+	// isTemporaryDial maintains the dial status
+	var isTemporaryDial bool
+
+	// if atlest one bootnode is connected the dial status is temporary
```

Comment:
Spelling error `atlest`

Reply:
commit-[link](https://github.com/0xPolygon/polygon-edge/pull/343/commits/aa333f9fcf309b004f3c1fbad82398cc7118a3b7)

Reply:
The correct spelling is `at least`

Reply:
Done, commit-[link](https://github.com/0xPolygon/polygon-edge/pull/343/commits/677053d94aec7557bc7d7778900ad03280f4caf2)